### PR TITLE
Profiler: Use sequential serial numbers for profiling events 

### PR DESCRIPTION
**Pull request by gunnarbeutner**
**Profiler: Use sequential serial numbers for profiling events**

Previously Profiler was using timestamps to distinguish processes. However it is possible that separate processes with the same PID exist at the exact same timestamp (e.g. for `execve`). This changes Profiler to use unique serial numbers for each event instead.

**Profiler: Remove m_deepest_stack_depth**

This isn't used anymore so let's remove it entirely.

### CODE REVIEW

File: Userland/DevTools/Profiler/EventSerialNumber.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#pragma once
+
+#include <AK/Assertions.h>
+#include <AK/Badge.h>
+#include <AK/NumericLimits.h>
+#include <AK/Types.h>
+
+namespace Profiler {
+
+// DistinctNumber's constructor is non-explicit which makes accidentally turning
```
Comment (linusg) [inline Userland/DevTools/Profiler/EventSerialNumber.h:16]: It's called `DistinctNumeric` though :)

File: Userland/DevTools/Profiler/EventSerialNumber.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+public:
+    constexpr EventSerialNumber()
+    {
+    }
+
+    constexpr explicit EventSerialNumber(size_t serial)
+        : m_serial(serial)
+    {
+        VERIFY(serial == 0);
+    }
```
Comment (linusg) [inline Userland/DevTools/Profiler/EventSerialNumber.h:28]: Hmm, if we don't allow any number other than zero then what even is the point of taking an argument here?

File: Userland/DevTools/Profiler/EventSerialNumber.h (hunk 1 above)
Comment (gunnarbeutner) [reply Userland/DevTools/Profiler/EventSerialNumber.h]: Fixed. :)

File: Userland/DevTools/Profiler/EventSerialNumber.h (hunk 2 above)
Comment (gunnarbeutner) [reply Userland/DevTools/Profiler/EventSerialNumber.h]: Yeah, that makes sense. I've removed the constructor and the explicit "{ 0 }" initializers. :+1:

**Pull request merged**